.scripts/Edit-Serialization.ps1

```diff
@@ -63,37 +63,55 @@ function Edit-InternalChatCompletionStreamResponseDeltaSerialization {
 
     # content serialization
     $inputRegex = @(
-        "if \(SerializedAdditionalRawData\?\.ContainsKey\(`"content`"\) != true && Optional\.IsDefined\(Content\)\)"
+        "if \(Optional\.IsDefined\(Content\) && _additionalBinaryDataProperties\?\.ContainsKey\(`"content`"\) != true\)"
     )
     $outputString = @(
         "// CUSTOM: Check inner collection is defined."
-        "if (SerializedAdditionalRawData?.ContainsKey(`"content`") != true && Optional.IsDefined(Content) && Content.IsInnerCollectionDefined())"
+        "if (Optional.IsDefined(Content) && _additionalBinaryDataProperties?.ContainsKey(`"content`") != true && Content.IsInnerCollectionDefined())"
     )
     Edit-Serialization -Filename $filename -InputRegex $inputRegex -OutputString $outputString -OutputIndentation 12
 
     # content deserialization
     $inputRegex = @(
         "return new InternalChatCompletionStreamResponseDelta\("
-        "    content,"
         "    functionCall,"
         "    toolCalls \?\? new ChangeTrackingList<StreamingChatToolCallUpdate>\(\),"
-        "    role,"
         "    refusal,"
-        "    serializedAdditionalRawData\);"
+        "    role,"
+        "    content,"
+        "    additionalBinaryDataProperties\);"
     )
     $outputString = @(
         "// CUSTOM: Initialize Content collection property."
         "return new InternalChatCompletionStreamResponseDelta("
-        "    content ?? new ChatMessageContent(),"
         "    functionCall,"
         "    toolCalls ?? new ChangeTrackingList<StreamingChatToolCallUpdate>(),"
-        "    role,"
         "    refusal,"
-        "    serializedAdditionalRawData);"
+        "    role,"
+        "    content ?? new ChatMessageContent(),"
+        "    additionalBinaryDataProperties);"
     )
     Edit-Serialization -Filename $filename -InputRegex $inputRegex -OutputString $outputString -OutputIndentation 12
 }
 
+function Edit-ChatMessageSerialization {
+    $filename = "ChatMessage.Serialization.cs"
+
+    # content serialization
+    $inputRegex = @(
+        "if \(true && Optional\.IsDefined\(Content\) && _additionalBinaryDataProperties\?\.ContainsKey\(`"content`"\) != true\)"
+    )
+    $outputString = @(
+        "// CUSTOM: Check inner collection is defined."
+        "if (true && Optional.IsDefined(Content) && Content.IsInnerCollectionDefined() && _additionalBinaryDataProperties?.ContainsKey(`"content`") != true)"
+    )
+    Edit-Serialization -Filename $filename -InputRegex $inputRegex -OutputString $outputString -OutputIndentation 12
+
+    # content deserialization
+    # no-op
+}
+
```

Comment:
Why are we adding updates using string comparison? Is this addressing an issue in code generation or TypeSpec?

FWIW, we should aim to remove these scripts altogether instead of adding more. If there's a gap in the generator or TypeSpec, fixing it at the source would be more maintainable and scalable.

Reply:
This is updating the existing post-generation scripts to account for the changes in code gen. I believe the goal is to remove these post-generation scripts in the future, but that is out of scope for this PR.

Reply:
Ah this is a net-new one. The reason is that serialization in MGC will delegate to the base type so the same transforms needed to be applied to the base type here.